experimental/InvariantTheory/AlgebraicGroups.jl

```diff
@@ -0,0 +1,218 @@
+import AbstractAlgebra.Field
+import AbstractAlgebra.Generic.MatSpaceElem
+
+
+export AlgGroup, subgroup, base_field, ambient_ring, defining_ideal, presentation_matrix, AlgGroup_GL, AlgGroup_SL
+
+export LinearRepresentation, representation_on_sym_power, mapping_matrix, group
+
+
+@Markdown.doc """
+    AlgGroup{FieldType, RingType, RingElemType, MatrixType}
+
+An algebraic group G defined as an algebraic subvariety of the space GL(t,𝕜). 
+The information stored consists of 
+  * the base field 𝕜 of type `FieldType`;
+  * a polynomial ring `R` = 𝕜[x₁,…,xₙ] of type `RingType`, 
+    the ambient polynomial ring for G seen as an affine variety;
+  * an ideal `I` ⊂ `R` of type `MPolyIdeal{RingElemType}` defining G 
+    as a subscheme of 𝔸ⁿ(𝕜);
+  * a t×t-matrix `Z` of type `MatrixType` with entries in `R` embedding 
+    G as an algebraic subscheme of GL(t,𝕜) and endowing it with its group structure.
+"""
+mutable struct AlgGroup{FieldType, RingType, RingElemType, MatrixType}
+  k::FieldType
+  R::RingType 			# The underlying polynomial ring
+  I::MPolyIdeal{RingElemType}	# The ideal describing the closed subgroup
+  Z::MatrixType			# The matrix coordinates for the representation
+
+  function AlgGroup( k::FieldType, R::RingType, I::MPolyIdeal{RingElemType}, Z::MatrixType) where{FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+    @assert typeof(Z) <: dense_matrix_type(R) "Matrix is not defined over the given ring"
+    @assert base_ring(R) === k "ambient polynomial ring is not defined over the given field" 
+    @assert MatrixType == dense_matrix_type(R) "Matrix has the wrong type"
+    return new{typeof(k), typeof(R), elem_type(R), dense_matrix_type(R)}(k, R, I, Z)
+  end
+end
+
+AlgGroup(R::RingType, I::MPolyIdeal{RingElemType}, Z) where{RingType<:Ring, RingElemType} = AlgGroup(base_ring(R), R, I, Z)
+
+function AlgGroup(I::MPolyIdeal{RingElemType}, Z) where{RingElemType}
+  @assert typeof(Z) == dense_matrix_type(base_ring(I)) 
+  @assert base_ring(Z) == base_ring(Z) "matrix is not defined over the same ring as the given ideal"
+  return AlgGroup(base_ring(base_ring(Z)), base_ring(Z), I, Z)
+end
+
+@Markdown.doc """
+    base_field(G::AlgGroup)
+
+Returns the field 𝕜 over which G is defined.
+"""
+function base_field(G::AlgGroup{FieldType, RingType, RingElemType, MatrixType}) where{
+	FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+  return G.k::FieldType
+end
```

Comment:
This is not idiomatic, and needlessly complicated/verbose (unless you plan to add `AlgGroup` variants with different parameter sets...). Just do this?
```suggestion
base_field(G::AlgGroup) = G.k
```
Obviously this applies a to a few more places.